storage/src/adb/any/mod.rs

```diff
@@ -2235,4 +2265,177 @@ pub(super) mod test {
             db.destroy().await.unwrap();
         });
     }
+
+    // Test that the`pruning_delay` works as expected.
```

Comment:
To ensure the committing changes are correct in adb::current, could you replicate these tests there?